From 3a4f755124e91ae763ad41b1e6b5fcb147fc7d12 Mon Sep 17 00:00:00 2001
From: Ren Jianing <jianing.ren@rock-chips.com>
Date: Mon, 8 Jun 2020 10:35:41 +0800
Subject: [PATCH] ARM: dts: rockchip: add vup_gpios in u2phy0 for rv11xx-ai-cam

The USB2.0 OTG PHY of RV1126/1109 which is designed for lower power
consumption provides only 8.8mA current source on DM. Multiplied
by 45 Ohm host termination resistance, voltage is about 400mV.
If the threshold voltage of host is greater than 400mV, the high
speed handshake will fail and SoC communicate at full speed. So
swing calibration is necessary.

We use gpio to control the 220 Ohm pull-up resistor to provide additional
current. Experiments show that the voltage of chirpK can be increases
to about 600mV.

Signed-off-by: Ren Jianing <jianing.ren@rock-chips.com>
Change-Id: I387ae1840959540ff9ca22db26896def0e890619
---
 arch/arm/boot/dts/rv11xx-ai-cam.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/rv11xx-ai-cam.dtsi b/arch/arm/boot/dts/rv11xx-ai-cam.dtsi
index 062753ed63e8..bc677f3e3f3d 100644
--- a/arch/arm/boot/dts/rv11xx-ai-cam.dtsi
+++ b/arch/arm/boot/dts/rv11xx-ai-cam.dtsi
@@ -323,6 +323,7 @@
 
 &u2phy0 {
 	status = "okay";
+	vup-gpios = <&gpio0 RK_PC1 GPIO_ACTIVE_LOW>;
 	u2phy_otg: otg-port {
 		status = "okay";
 		rockchip,vbus-always-on;
-- 
2.35.3

